src/ui/App.tsx

```diff
@@ -1,13 +1,37 @@
-import NotebookFooter from "./NotebookFooter";
-import NotebookHeader from "./NotebookHeader";
-import NotebookView from "./NotebookView";
+import { useEffect } from "react";
+import { Route, Router, Switch } from "wouter";
+import { useHashLocation } from "wouter/use-hash-location";
+
+import Home from "~/pages/Home";
+import Notebook from "~/pages/Notebook";
+
+type Props = {
+  openedFile?: string;
+};
+
+export default function App({ openedFile }: Props) {
+  const [, navigate] = useHashLocation();
+
+  // if we're opened with a file, we should navigate to the notebook page
+  useEffect(() => {
+    if (openedFile) {
+      navigate(`/notebook/${encodeURIComponent(openedFile)}`);
```

Comment:
lgtm! I was actually wondering if it's possible to open this url directly from rust so you didn't have to send it over to via the global variable